src/undo_log/undo_log.h

```diff
@@ -20,106 +21,8 @@
 // operations within the undo log to rollback the graph to its
 // original state
 
-// UndoLog operation types
-typedef enum {
-	UNDO_UPDATE = 0,     // undo entity update
-	UNDO_CREATE_NODE,    // undo node creation
-	UNDO_CREATE_EDGE,    // undo edge creation
-	UNDO_DELETE_NODE,    // undo node deletion
-	UNDO_DELETE_EDGE,    // undo edge deletion
-	UNDO_SET_LABELS,     // undo set labels
-	UNDO_REMOVE_LABELS,  // undo remove labels
-	UNDO_ADD_SCHEMA,     // undo schema addition
-	UNDO_ADD_ATTRIBUTE,  // undo property addition
-	UNDO_CREATE_INDEX,   // undo index addition
-} UndoOpType;
-
-//------------------------------------------------------------------------------
-// Undo operations
-//------------------------------------------------------------------------------
-
-// undo node/edge creation
-typedef struct UndoCreateOp UndoCreateOp;
-struct UndoCreateOp {
-	union {
-		Node n;
-		Edge e;
-	};
-};
-
-// undo node deletion
-typedef struct UndoDeleteNodeOp UndoDeleteNodeOp;
-struct UndoDeleteNodeOp {
-	EntityID id;
-	AttributeSet set;
-	LabelID *labels;   // labels attached to deleted entity
-	uint label_count;  // number of labels attached to deleted entity
-};
-
-// undo edge deletion
-typedef struct UndoDeleteEdgeOp UndoDeleteEdgeOp;
-struct UndoDeleteEdgeOp {
-	EntityID id;
-	int relationID;          // Relation ID
-	NodeID src_id;           // Source node ID
-	NodeID dest_id;          // Destination node ID
-	AttributeSet set;
-};
-
-// undo graph entity update
-typedef struct UndoUpdateOp UndoUpdateOp;
-struct UndoUpdateOp {
-	union {
-		Node n;
-		Edge e;
-	};
-	GraphEntityType entity_type;  // node/edge
-	AttributeSet set;             // old attribute set
-};
-
-typedef struct UndoLabelsOp UndoLabelsOp;
-struct UndoLabelsOp {
-	Node node;
-	LabelID* label_ids;
-	ushort labels_count;
-};
-
-typedef struct UndoAddSchemaOp UndoAddSchemaOp;
-struct UndoAddSchemaOp {
-	int schema_id;
-	SchemaType t;
-};
-
-typedef struct UndoAddAttributeOp UndoAddAttributeOp;
-struct UndoAddAttributeOp {
-	Attribute_ID attribute_id;
-};
-
-typedef struct UndoCreateIndexOp UndoCreateIndexOp;
-struct UndoCreateIndexOp {
-	SchemaType st;      // schema type
-	const char *label;  // label / relationship
-	const char *field;  // field
-	IndexFieldType t;   // type of index
-};
-
-// Undo operation
-typedef struct {
-	union {
-		UndoCreateOp create_op;
-		UndoDeleteNodeOp delete_node_op;
-		UndoDeleteEdgeOp delete_edge_op;
-		UndoUpdateOp update_op;
-		UndoLabelsOp labels_op;
-		UndoAddSchemaOp schema_op;
-		UndoAddAttributeOp attribute_op;
-		UndoCreateIndexOp index_op;
-	};
-	UndoOpType type;  // type of undo operation
-} UndoOp;
-
 // container for undo_list
-typedef DataBlock *UndoLog;
+typedef void *UndoLog;
```

Comment:
I think it would be more in line with C to declare UndoLog as an opaque like so:
```c
struct Opaque_UndoLog
typedef struct Opaque_UndoLog *UndoLog;
```